app/helpers/request_helper.py

```diff
@@ -106,8 +106,5 @@ def get_doc_from_store(tx_id):
     response = remote_call(store_url)
 
     if response_ok(response):
-        logger.info(
-            "Successfully got document from store",
-            tx_id=tx_id,
-        )
+        logger.info("Successfully got document from store", tx_id=tx_id)
```

Comment:
We should be binding at the entry to the application therefore we shouldn't need to add `tx_id` anywhere else?

Reply:
I removed it and now it's not logging out the tx_id.  I don't really want to address this comment as all I did was tidy up a line that was split over multiple lines.
Could you approve it? as the PR is doing what it's meant to, even if the rest of the code around it isn't perfect.

Reply:
It may seem petty and I hate to agree with Jon but this PR is about binding tx_id to the logging so removing the explicit declaration of tx_id from here and anywhere similar could easily be classed as part of this PR. Then also take boy scout rule then it's probably best to remove it. I'm confused why tx_id wouldn't show on this log message if you remove it though if the earlier binding is working as expected everywhere else.

Reply:
You both make it sound so easy... I've spent some time on it and I **think** I know what's up, but don't know how to sort it whilst making it tidy at the same time.
So the logger (pythons stdlib one) is common between all modules which is why the formatting stays.  Every call to `logging.getLogger(__name__)` gives you the same formatted logger.
The result of `wrap_logger(logging.getLogger(__name__))` gives you a fresh structured logger each time.  I can't find a way to keep the binding between all the different modules without literally passing the bound logger between every class.  It kinda works for `message_processor.py` and `sdxftp.py` because  message processor makes a logger, binds tx_id to it, then passes that bound logger to `sdxftp.py` the problem comes when the `request_helper.py` is used.  It's not a class, it's just a bunch of helper functions, so (again, I think...) when it's invoked it creates its own structured logger that has loggers formatting, but has nothing bound to it because it's new.

Any thoughts about this?  Is there a way to created a structured logger with bindings that can be easily used between multiple locations?  If not, can we just accept that `request_helper.py` will have explicit tx_id fields in its log lines?
I don't want to spend more time on this as it was meant to be a very quick job to make life easier, not a thorough look at how downstream handles its logging.

Reply:
@insacuri - Might be worth trying using structlog's `get_logger` method rather than `wrap_logger`, that should give you a global context for the binds. 
I don't know if this would cause issues between (concurrent?) requests though.

Reply:
If above doesn't work then i'm happy to accept it as if. I've already ticked it so convince @jonnyshaw89.

Reply:
Got it working after a bunch of hassle.  Had to change a lot more files than I was intending to.
@dwyeradam can you retest it because I've touched a bunch more files since you did your test.
@jonnyshaw89 can you take another looksee?